spring-kafka/src/test/java/org/springframework/kafka/listener/AbstractConsumerSeekAwareTests.java

```diff
@@ -0,0 +1,132 @@
+
+	@Test
+	public void seekForAllGroups() throws Exception {
+		template.send(TOPIC, "test-data");
+		template.send(TOPIC, "test-data");
+		assertTrue(MultiGroupListener.latch1.await(10, TimeUnit.SECONDS));
+		assertTrue(MultiGroupListener.latch2.await(10, TimeUnit.SECONDS));
+
+		MultiGroupListener.latch1 = new CountDownLatch(2);
+		MultiGroupListener.latch2 = new CountDownLatch(2);
+
+		multiGroupListener.seekToBeginning();
+		assertTrue(MultiGroupListener.latch1.await(10, TimeUnit.SECONDS));
+		assertTrue(MultiGroupListener.latch2.await(10, TimeUnit.SECONDS));
+	}
+
+	@Test
+	public void seekForSpecificGroup() throws Exception {
+		template.send(TOPIC, "test-data");
+		template.send(TOPIC, "test-data");
+		assertTrue(MultiGroupListener.latch1.await(10, TimeUnit.SECONDS));
+		assertTrue(MultiGroupListener.latch2.await(10, TimeUnit.SECONDS));
+
+		MultiGroupListener.latch1 = new CountDownLatch(2);
+		MultiGroupListener.latch2 = new CountDownLatch(2);
+
+		multiGroupListener.seekToBeginningForGroup("group2");
+		assertThat(MultiGroupListener.latch1.getCount()).isEqualTo(2);
+		assertTrue(MultiGroupListener.latch2.await(10, TimeUnit.SECONDS));
+	}
+
+	@EnableKafka
+	@Configuration
+	static class Config {
+
+		@Autowired
+		EmbeddedKafkaBroker broker;
+
+		@Bean
+		public ConcurrentKafkaListenerContainerFactory<String, String> kafkaListenerContainerFactory(
+				ConsumerFactory<String, String> consumerFactory) {
+			ConcurrentKafkaListenerContainerFactory<String, String> factory = new ConcurrentKafkaListenerContainerFactory<>();
+			factory.setConsumerFactory(consumerFactory);
+			return factory;
+		}
+
+		@Bean
+		ConsumerFactory<String, String> consumerFactory() {
+			return new DefaultKafkaConsumerFactory<>(KafkaTestUtils.consumerProps("test-group", "false", this.broker));
+		}
+
+		@Bean
+		ProducerFactory<String, String> producerFactory() {
+			return new DefaultKafkaProducerFactory<>(KafkaTestUtils.producerProps(this.broker));
+		}
+
+		@Bean
+		KafkaTemplate<String, String> template(ProducerFactory<String, String> pf) {
+			return new KafkaTemplate<>(pf);
+		}
+
+		@Component
+		static class MultiGroupListener extends AbstractConsumerSeekAware {
+
+			static CountDownLatch latch1 = new CountDownLatch(2);
+			static CountDownLatch latch2 = new CountDownLatch(2);
+
+			@KafkaListener(groupId = "group1", topics = TOPIC)
+			void listenForGroup1(String in) {
+//				System.out.printf("[group1] in = %s\n", in); // TODO remove
+				latch1.countDown();
+			}
+
+			@KafkaListener(groupId = "group2", topics = TOPIC)
+			void listenForGroup2(String in) {
+//				System.out.printf("[group2] in = %s\n", in); // TODO remove
+				latch2.countDown();
+			}
+
+			public void seekToBeginningForGroup(String groupIdForSeek) {
+				getCallbacksAndTopics().forEach((cb, topics) -> {
+					if (groupIdForSeek.equals(cb.getGroupId())) {
+						topics.forEach(tp -> cb.seekToBeginning(tp.topic(), tp.partition()));
+					}
+				});
+			}
```

Comment:
I also noticed something awkward when inserting values into `AbstractConsumerSeekAware.callbacks` using `onPartitionsAssigned()`, and I'm not sure if this was intentional. 

Currently, `TopicPartition` serves as the key and `ConsumerSeekCallback` as the value. However, if there are different group IDs, a `TopicPartition` should be allowed to have multiple callbacks. This relates to why I chose to use `getCallbacksAndTopics()` instead of `getSeekCallbacks()`. 

While this issue isn't addressed in the current PR, we can discuss it in another issue or PR. In summary, a beneficial change would be as follows:
Current: `Map<TopicPartition, ConsumerSeekCallback> callbacks`
Proposed: `Map<TopicPartition, List<ConsumerSeekCallback>> callbacks`

Reply:
Please take a look this later. https://github.com/spring-projects/spring-kafka/issues/3328